src/google/protobuf/port_def.inc

```diff
@@ -413,6 +413,26 @@
 #define PROTOBUF_ATTRIBUTE_REINITIALIZES
 #endif
 
+#ifdef PROTOBUF_NO_RETURN
+#error PROTOBUF_NO_RETURN was previously defined
+#endif
+#ifdef __cpp_has_attribute
```

Comment:
the prelud of pord_def.inc (around linke 55) allows you to skip a lot of this and implement this as

#if __has_cpp_attribute(noreturn)
#elfi ...